packages/core/src/resolvers/arguments.ts

```diff
@@ -1,30 +1,33 @@
 import sift, { generateFilterSetManifest } from '../utils/sift';
-import {
-  resolveMaybeThunk,
-  SchemaComposer,
-  ThunkComposer,
-} from 'graphql-compose';
+import { ContentNode, FlatbreadConfig } from 'flatbread';
+import { FlatbreadProvider } from '../providers/base';
+
+interface ResolveQueryArgsOptions {
+  type: {
+    name: string;
+    pluralName: string;
+    pluralQueryName: string;
+  };
+}
 
 /**
  * Resolvers for query arguments.
  */
 const resolveQueryArgs = async (
   nodes: any[],
   args: any,
-  type: string,
-  schemaComposer: SchemaComposer
+  config: FlatbreadConfig,
+  options: ResolveQueryArgsOptions
 ) => {
   const { skip, limit, order, sortBy, filter } = args;
 
-  // console.group('filter middleware');
-  // console.log('args', args);
-  // console.log('type', type);
-  // console.log('scoped result', nodes);
-  // console.groupEnd();
-
   if (filter) {
-    // nodes = await resolveFilter(filter, nodes, type, schemaComposer);
-    nodes = nodes.filter(sift(filter));
+    // Turn the filter into a GraphQL subquery that returns an array of matching content node IDs.
+    const listOfNodeIDsToFilter = await resolveFilter(filter, config, options);
+
+    nodes = listOfNodeIDsToFilter.map((desiredNodeId) =>
+      nodes.find((node) => node.id === desiredNodeId)
+    );
```

Comment:
it might be worth storing nodes in a map here to cut down on loops
```suggestion

    const nodeById = keyBy(nodes, 'id');
    nodes = listOfNodeIdsToFilter.map(derisedNodeId => nodeById[desiredNodeId])
```

Reply:
Great suggestion!